rust/fastsim-core/src/traits.rs

```diff
@@ -154,6 +155,44 @@ pub trait SerdeAPI: Serialize + for<'a> Deserialize<'a> {
     fn from_bincode(encoded: &[u8]) -> anyhow::Result<Self> {
         Ok(bincode::deserialize(encoded)?)
     }
+
+    /// instantiates an object from a url
+    fn from_url<S: AsRef<str>>(url: S) -> anyhow::Result<Self> {
+        let url = url.as_ref();
+        let temp_dir = tempdir()?;
+        let mut file_path = PathBuf::new();
+        // do these file types need to be specific to the object?
+        // TODO: either make funciton work for csv files, or remove from supported file list
+        if url.ends_with("yaml"){
+            let file_path = temp_dir.path().join("temporary_object.yaml");
+        } else if url.ends_with("csv"){
+            let file_path = temp_dir.path().join("temporary_object.csv");
+        } else if url.ends_with("json"){
+            let file_path = temp_dir.path().join("temporary_object.json");
+        } else {
+            bail!("Unsupported file type, must be a yaml, json, or csv file.");
```

Comment:
Make this bail message look more like what's here:
https://github.com/NREL/fastsim/blob/5d2c1d4b331c58a87c9bd390e1ff6005bc22e8ca/rust/fastsim-core/src/traits.rs#L73